synthesize/src/main.rs

```diff
@@ -0,0 +1,46 @@
+use std::io::{self, Write};
+use std::path::PathBuf;
+
+use synthesize::Result;
+
+fn main() -> Result<()> {
+    if let Err(e) = self::cli() {
+        writeln!(io::stderr(), "{}", e)?;
```

Comment:
```suggestion
    if let Err(error) = self::cli() {
        eprintln!("{error}");
```

Reply:
I used `writeln` because it can propagate the error in writing instead of panicking: https://github.com/BurntSushi/advent-of-code/issues/17.